kategory-docs/docs/docs/datatypes/try/README.md

```diff
@@ -4,6 +4,182 @@ title: Try
+
+```kotlin:ank
+open class GeneralException: Exception()
+
+class NoConnectionException: GeneralException()
+
+class AuthorizationException: GeneralException()
+
+fun checkPermissions() {
+    throw AuthorizationException()
+}
+
+fun getLotteryNumbersFromCloud(): List<String> {
+    throw NoConnectionException()
+}
+
+fun getLotteryNumbers(): List<String> {
+    checkPermissions()
+    
+    return getLotteryNumbersFromCloud()
+}
+```
+
+The traditional way to control this would be to use a `try-catch` block, as we have said before:
+
+```kotlin:ank
+try {
+    getLotteryNumbers()
+} catch (e: NoConnectionException) {
+    println("You don't have connection")
+} catch (e: AuthorizationException) {
+    println("You don't have permission")
+}
+```
+
+However, we could use `Try` to retrieve the computation result in a much cleaner way:
+
+```kotlin:ank
+import kategory.*
+
+val lotteryTry = Try { getLotteryNumbers() }
+lotteryTry
+```
+
+By using `getOrElse` we can give a default value to return, when the computation fails, similar to what we can also do with `Option` when there is no value:
+
+```kotlin:ank
+lotteryTry.getOrElse { emptyList() }
+```
+
+If you want to perform a check on a possible success, you can use `filter` to convert successful computations in failures if conditions aren't met:
+
+```kotlin:ank
+lotteryTry.filter {
+    it.size < 4
+}
+```
+
+We can also use `recover` which allow us to recover from a particular error (we receive the error and have to return a new value):
+
+```kotlin:ank
+lotteryTry.recover { exception ->
+    emptyList()
+}
+```
+
+Or if you have another different computation that can also fail, you can use `recoverWith` to recover from an error (as you do with `recover`, but in this case, returning a new `Try`):
+
+```kotlin:ank
+enum class Source {
+    CACHE, NETWORK
+}
+
+fun getLotteryNumbers(source: Source): List<String> {
+    checkPermissions()
+
+    return getLotteryNumbersFromCloud()
+}
+
+Try { getLotteryNumbers(Source.NETWORK) }.recoverWith {
+    Try { getLotteryNumbers(Source.CACHE) }
+}
+```
+
+When you want to handle both cases of the computation you can use `fold`. With `fold` we provide two functions, one for transforming a failure into a new value, the second one to transform the success value into a new one:
+
+```kotlin:ank
+lotteryTry.fold(
+    { emptyList<String>() },
+    { it.filter { it.toIntOrNull() != null } })
+```
+
+Or, as we have with `recoverWith`, we can use a version of `fold` which allows us to handle both cases with functions that return a new instance of `Try`, `transform`:
+
+```kotlin:ank
+lotteryTry.transform(
+    { Try { it.map { it.toInt() } } },
+    { Try.pure(emptyList<Int>()) })
+```
+
+Lastly, Kategory contains `Try` instances for many useful typeclasses that allows you to use and transform fallibale values:
+
+[`Functor`]({{ '/docs/typeclasses/functor/' | relative_url }})
+
+Transforming the value, if the computation is a success
```

Comment:
Very minor, but for consistancy this heading, `Computing over independent values`, and`Computing over dependent values ignoring failure` should end in a colon. e.g.   Transforming the value, if the computation is a success:

Reply:
https://github.com/kategory/kategory/pull/458/commits/2574a1a7f173591fc076ad3686a271eff2d79546 Thanks :)